### PR TITLE
:bug: (deployimage/v1-beta1): fix scaffold with multigroup and and add optional plugins to multigroup sample to validate changes

### PR DESCRIPTION
**Pull request by camilamacedo86**
## Description
Add the fix for: https://github.com/kubernetes-sigs/kubebuilder/issues/3558
Also, add call to grafana and deployimage into the multigroup sample so that we can either test this options with multigroup layout

Closes:  https://github.com/kubernetes-sigs/kubebuilder/issues/3558

### CODE REVIEW

**Comment by camilamacedo86** [comment]
Hi @yyy1000,

Here is the fix for the issue: https://github.com/kubernetes-sigs/kubebuilder/issues/3558

@Kavinjsir @varshaprasad96 

I am moving forward with this one for we do not block the generate alpha command project.

**Review by yyy1000 — COMMENTED** [review]
LGTM!

File: pkg/plugins/golang/deploy-image/v1alpha1/scaffolds/api.go [hunk 1]
```diff
@@ -189,28 +189,15 @@ func (s *apiScaffolder) scaffoldCreateAPIFromPlugins(isLegacyLayout bool) error
 // which will have its own controller template which set the recorder so that we can use it
 // in the reconciliation to create an event inside for the finalizer
 func (s *apiScaffolder) updateMainByAddingEventRecorder(isGoV3 bool, defaultMainPath string) error {
```
Comment (Kavinjsir) [inline pkg/plugins/golang/deploy-image/v1alpha1/scaffolds/api.go:191]: Wondering if we can drop the parameter `isGoV3 bool` if it is no longer used?

**Review by Kavinjsir — CHANGES_REQUESTED** [review]

File: pkg/plugins/golang/deploy-image/v1alpha1/scaffolds/api.go (hunk 1 above)
Comment (camilamacedo86) [reply pkg/plugins/golang/deploy-image/v1alpha1/scaffolds/api.go]: Yep good point.
I changed that

**Comment by camilamacedo86** [comment]
@Kavinjsir 

Could we move forward now?
Have we your LGTM as well?

**Review by Kavinjsir — APPROVED** [review]
LGTM!

**Pull request merged**